rust/operator/src/kerberos.rs

```diff
@@ -22,80 +35,174 @@ impl HdfsSiteConfigBuilder {
+                .add("dfs.datanode.keytab.file", "/stackable/kerberos/keytab")
                 .add(
                     "dfs.journalnode.kerberos.principal.pattern",
-                    // E.g. jn/hdfs-test-journalnode-default-0.hdfs-test-journalnode-default.test.svc.cluster.local@CLUSTER.LOCAL
-                    format!("jn/{hdfs_name}-journalnode-*.{hdfs_name}-journalnode-*.{hdfs_namespace}.svc.cluster.local@${{env.KERBEROS_REALM}}").as_str(),
+                    format!("jn/{principal_host_part}"),
                 )
                 .add(
                     "dfs.namenode.kerberos.principal.pattern",
-                    format!("nn/{hdfs_name}-namenode-*.{hdfs_name}-namenode-*.{hdfs_namespace}.svc.cluster.local@${{env.KERBEROS_REALM}}").as_str(),
+                    format!("nn/{principal_host_part}"),
+                )
+                // Otherwise we fail with `java.io.IOException: No groups found for user nn`
+                // Default value is `dr.who=`, so we include that here
+                .add(
+                    "hadoop.user.group.static.mapping.overrides",
+                    "dr.who=;nn=;nm=;jn=;",
                 );
 
-            match role {
-                HdfsRole::NameNode => {
-                    self.add(
-                        "dfs.namenode.kerberos.principal",
-                        "nn/_HOST@${env.KERBEROS_REALM}",
-                    )
-                    .add("dfs.namenode.keytab.file", "/stackable/kerberos/keytab");
-                }
-                HdfsRole::DataNode => {
-                    self.add(
-                        "dfs.datanode.kerberos.principal",
-                        "dn/_HOST@${env.KERBEROS_REALM}",
-                    )
-                    .add("dfs.datanode.keytab.file", "/stackable/kerberos/keytab");
-                }
-                HdfsRole::JournalNode => {
-                    self.add(
-                        "dfs.journalnode.kerberos.principal",
-                        "jn/_HOST@${env.KERBEROS_REALM}",
-                    )
-                    .add("dfs.journalnode.keytab.file", "/stackable/kerberos/keytab")
-                    .add(
-                        "dfs.journalnode.kerberos.internal.spnego.principal",
-                        "HTTP/_HOST@${env.KERBEROS_REALM}",
-                    );
-                }
-            }
+            self.add_wire_encryption_settings();
         }
         self
     }
 
     pub fn security_discovery_config(&mut self, hdfs: &HdfsCluster) -> &mut Self {
         if hdfs.has_kerberos_enabled() {
-            self.add(HADOOP_SECURITY_AUTHENTICATION, "kerberos");
+            self.add("hadoop.security.authentication", "kerberos");
+            self.add_wire_encryption_settings();
         }
         self
     }
+
+    fn add_wire_encryption_settings(&mut self) -> &mut Self {
+        self.add("hadoop.rpc.protection", "privacy");
+        self
+    }
 }
+
+// IMPORTANT: We only support Kerberos for HDFS >= 3.3.x
```

Comment:
Can we move this section up into the `check_if_supported` fn? Maybe with the full stack trace snipped out or moved into a gist/separate file.

Reply:
277c5503ffae6962acd3e55020dff9b0edc8194a